addon-test-support/@ember/test-helpers/settled.ts

```diff
@@ -142,12 +143,43 @@ function checkWaiters() {
   return false;
 }
 
+let currentPauseTestPromiseSemaphore = 0;
+
+/**
+  Ensures that tests wait for the provided promise before being considered "settled".
+
+  This function is expected to be used in application/addon code (aka non-test code).
+  When used during a test run, it ensures that the next `await settled()` check does
+  not result until after the promise has settled, when used outside of a test it is a
+  simple "pass through" function.
+
+  @public
+  @param {Promise<T>} promise the promise to wait for
+  @returns {Promise<T>} the passed in promise
+*/
+export function pauseTestFor<T>(promise: Promise<T>): Promise<T> {
+  currentPauseTestPromiseSemaphore++;
+  return promise.then(
+    result => {
+      currentPauseTestPromiseSemaphore--;
+
+      return result;
+    },
+    error => {
```

Comment:
Why this vs. `catch`?

Reply:
```js
promise
  .then(...)
  .catch(...)
```

Creates 2 chained promises, whereas `promise.then(good,bad)` makes one.